src/lib/services/RuleEngine.ts

```diff
@@ -50,3 +56,35 @@ export interface RuleEngine {
 	 */
 	isEngineRequested(filterValues: string[], engineOptions: Map<string, string>): boolean;
 }
+
```

Comment:
AbstractRuleEngine is used to create a "wrapper" function for run called runHighLevel that is the same across engines. The function calls the engines run function then uses normalizeSeverity to parse the results and update severities. Now the engines extend this class instead of implementing RuleEngine.

Reply:
I'm not in love with the name `runHighLevel`, because it's not entirely clear what that means. Can we rename it something more clear?

Reply:
How about `runEngine` that later calls the existing `run`? If they had different signatures, the same name could've been reused.